src/blend.cpp

```diff
@@ -315,81 +312,106 @@ void Work_Blend(uv_work_t* req)
         if (!alpha) break;
         auto image = *rit;
         if (!image) continue;
-        std::unique_ptr<mapnik::image_reader> image_reader;
-        try {
-            image_reader = std::unique_ptr<mapnik::image_reader>(mapnik::get_image_reader(image->data, image->dataLength));
-        } catch (std::exception const& ex) {
-            baton->message = ex.what();
-            return;
-        }
 
-        if (!image_reader || !image_reader.get()) {
-            // Not quite sure anymore how the pointer would not be returned
-            // from the reader and can't find a way to make this fail.
-            // So removing from coverage
-            /* LCOV_EXCL_START */
-            baton->message = "Unknown image format";
-            return;
-            /* LCOV_EXCL_STOP */
-        }
+        if (image->im_obj) {
+            unsigned layer_width = image->im_obj->get()->width();
+            unsigned layer_height = image->im_obj->get()->height();
+            if (layer_width == 0 || layer_height == 0) {
+                baton->message = "zero width/height image encountered";
+                return;
+            }
+            int visibleWidth = (int)layer_width + image->x;
```

Comment:
nitpick: c++ casts not c-style cases. We really should be using `-Wold-style-cast` here to catch that.

Reply:
@springmeyer we probably should, this was copied or indented code -- I am worried how much this will light up in the entire code base though.

Reply:
Ah, okay - it was not clear to be this was indented. Feel free to ignore my comment then to keep this focused.